shared/settings/delete-confirm/index.desktop.js

```diff
@@ -1,33 +1,58 @@
 // @flow
-import React from 'react'
+import React, {Component} from 'react'
 import {globalStyles, globalMargins, globalColors} from '../../styles'
 import {Box, Button, Icon, Text, Avatar} from '../../common-adapters'
 
 import type {Props} from './index'
 
-function DeleteConfirm (props: Props) {
-  return (
-    <Box style={{...globalStyles.flexBoxColumn, alignItems: 'center', justifyContent: 'center', flex: 1}}>
-      <Avatar
-        size={48}
-        username={props.username}>
-        <Icon type='iconfont-remove' style={iconStyle} />
-      </Avatar>
-      <Text type='HeaderError' style={{...globalStyles.italic, textDecoration: 'line-through'}}>{props.username}</Text>
-      <Text type='Header' style={{marginTop: globalMargins.medium, width: 320, textAlign: 'center'}}>Are you sure you want to permanently delete your account?</Text>
-      <Box style={{...globalStyles.flexBoxRow, marginTop: globalMargins.medium}}>
-        <Button
-          type='Secondary'
-          label='Cancel'
-          onClick={props.onCancel} />
-        <Button
-          style={{margin: 0}}
-          type='Danger'
-          label='Yes, permanently delete it'
-          onClick={props.onDeleteForever} />
+type State = {
+  allowDeleteForever: boolean,
+}
+
+class DeleteConfirm extends Component<void, Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props)
+
+    this.state = {allowDeleteForever: false}
+  }
+
+  componentWillMount () {
+    this.setState({allowDeleteForever: false})
+  }
+
+  componentDidMount () {
+    setTimeout(() => {
+      this.setState({allowDeleteForever: true})
```

Comment:
Will this throw if the component is destroyed before the `setTimeout` fires? Alex created https://github.com/keybase/client/blob/master/shared/common-adapters/hoc-timers.js to ensure that `clearTimeout` is called when a component is unmounted, might be a use case here?